dvc/external_repo.py

```diff
@@ -31,7 +32,7 @@ def external_repo(url, rev=None, for_write=False):
     logger.debug("Creating external repo %s@%s", url, rev)
     path = _cached_clone(url, rev, for_write=for_write)
     if not rev:
-        rev = "HEAD"
+        rev = "refs/remotes/origin/HEAD"
```

Comment:
Since `rev=None` is supposed to mean tip on the default branch we want to use HEAD from the origin repo, not HEAD from our local clone working copy. Local HEAD points to the tip of whatever branch we first cloned from (which may not be the default branch), but here we want the tip of the default branch.

Reply:
I would be happy to see such things commented about in dvc code.